src/openrct2/world/Map.cpp

```diff
@@ -762,26 +761,25 @@ void MapUpdatePathWideFlags()
     // Presumably update_path_wide_flags is too computationally expensive to call for every
     // tile every update, so gWidePathTileLoopX and gWidePathTileLoopY store the x and y
     // progress. A maximum of 128 calls is done per update.
-    auto x = gWidePathTileLoopPosition.x;
-    auto y = gWidePathTileLoopPosition.y;
+    auto& gameState = GetGameState();
+    TileCoordsXY loopPosition = gameState.WidePathTileLoopPosition;
     for (int32_t i = 0; i < 128; i++)
     {
-        FootpathUpdatePathWideFlags({ x, y });
+        FootpathUpdatePathWideFlags(loopPosition.ToCoordsXY());
```

Comment:
This looks weird, previously this conversion from TileCoordsXY to CoordsXY was invisible, but now it stands out. Maybe we should add a `TODO` to investigate whether. Maybe even `WidePathTileLoopPosition` should be `CoordsXY`, since that's the only context it is ever used for, even with `COORDS_XY_STEP`. What do you think @Gymnasiast ?

Reply:
I think for tile coordinates we should use the TileCoords type, having everything CoordsXY often requires to iterate it by incrementing by tile size which makes the code a bit more complicated.

Reply:
Ugh, this code is incredibly dirty. It uses `TileCoordsXY` as its type, yet is used as if it’s a CoordsXY, and seemingly also using those big coordinates in S4, S6 and Park.

While I agree with Matt that `TileCoordsXY` would have been a better fit, I think the best solution (at least) for now is to change the declared data type to `CoordsXY` to bring it in line with what actually saved in it. If a refactor/change of data type is desired, that would be better left to another PR.

@jan-kelemen The changes you need to make is changing the type of WidePathTileLoopPosition in GameState_t to `CoordsXY`, and changing above call to `FootpathUpdatePathWideFlags(loopPosition);`

Reply:
Type changed to `CoordsXY`